compiler/src/typed/typecore.re

```diff
@@ -391,6 +448,124 @@ let primn_type =
         ("length", Builtin_types.type_wasmi32),
       ],
       Builtin_types.type_wasmi32,
+    )
+  | WasmTernaryV128(_) =>
+    prim_type(
+      [
+        ("vec1", Builtin_types.type_wasmv128),
+        ("vec2", Builtin_types.type_wasmv128),
+        ("vec3", Builtin_types.type_wasmv128),
+      ],
+      Builtin_types.type_wasmv128,
+    )
+  | WasmSimdReplace({wasm_op, arg_type}) =>
+    prim_type(
+      [
+        ("vec", Builtin_types.type_wasmv128),
+        ("lane", Builtin_types.type_wasmi32),
+        ("value", grain_type_of_wasm_prim_type(arg_type)),
+      ],
+      Builtin_types.type_wasmv128,
+    )
+  | WasmSimdShuffle =>
+    prim_type(
+      [
+        ("vec1", Builtin_types.type_wasmv128),
+        ("vec2", Builtin_types.type_wasmv128),
+        ("lane0Idx", Builtin_types.type_wasmi32),
+        ("lane1Idx", Builtin_types.type_wasmi32),
+        ("lane2Idx", Builtin_types.type_wasmi32),
+        ("lane3Idx", Builtin_types.type_wasmi32),
+        ("lane4Idx", Builtin_types.type_wasmi32),
+        ("lane5Idx", Builtin_types.type_wasmi32),
+        ("lane6Idx", Builtin_types.type_wasmi32),
+        ("lane7Idx", Builtin_types.type_wasmi32),
+        ("lane8Idx", Builtin_types.type_wasmi32),
+        ("lane9Idx", Builtin_types.type_wasmi32),
+        ("lane10Idx", Builtin_types.type_wasmi32),
+        ("lane11Idx", Builtin_types.type_wasmi32),
+        ("lane12Idx", Builtin_types.type_wasmi32),
+        ("lane13Idx", Builtin_types.type_wasmi32),
+        ("lane14Idx", Builtin_types.type_wasmi32),
+        ("lane15Idx", Builtin_types.type_wasmi32),
+      ],
+      Builtin_types.type_wasmv128,
+    )
+  | WasmSimdLoad({wasm_op}) =>
+    prim_type(
+      [
+        ("ptr", Builtin_types.type_wasmi32),
+        ("offset", Builtin_types.type_wasmi32),
+        ("align", Builtin_types.type_wasmi32),
+      ],
+      Builtin_types.type_wasmv128,
+    )
+  | WasmSimdLoadStoreLane({wasm_op, ret_type}) =>
+    prim_type(
+      [
+        ("ptr", Builtin_types.type_wasmi32),
+        ("offset", Builtin_types.type_wasmi32),
+        ("align", Builtin_types.type_wasmi32),
+        ("lane", Builtin_types.type_wasmi32),
+        ("vec", Builtin_types.type_wasmv128),
+      ],
+      grain_type_of_wasm_prim_type(ret_type),
+    )
+  | WasmSimdConstI8x16 =>
+    prim_type(
+      [
+        ("lane0", Builtin_types.type_wasmi32),
```

Comment:
I wonder if it would have been worth using a helper like `List.init((index) => ("lane" ++ toString(index),Builtin_types.type_wasmi32), 15)` for all of these.